include/bio/ranges/views/add_reverse_complement.hpp

```diff
@@ -0,0 +1,256 @@
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/biocpp/biocpp-core/blob/main/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides bio::views::add_reverse_complement.
+ */
+
+#pragma once
+
+#include <ranges>
+
+#include <bio/alphabet/nucleotide/concept.hpp>
+#include <bio/ranges/views/deep.hpp>
+#include <bio/ranges/views/transform_by_pos.hpp>
+
+// ============================================================================
+//  reverse_complement_or_not_fn (adaptor definition)
+// ============================================================================
+
+namespace bio::ranges::detail
+{
+
+/*!\brief Definition of the range adaptor object type for bio::views::reverse_complement_or_not.
+ */
+struct reverse_complement_or_not_fn
+{
+private:
+    //!\brief Return original nucleotide at a given position.
+    static auto func_nop(auto && urange, size_t pos) { return urange[pos]; }
+
+    //!\brief Return nucleotide of reverse complemented range at a given position.
+    static auto func_revcomp(auto && urange, size_t pos)
+    {
+        return bio::alphabet::complement(urange[std::ranges::size(urange) - pos - 1]);
+    }
+
+public:
+    //!\brief Store the argument and return a range adaptor closure object.
+    constexpr auto operator()(bool const transform) const { return detail::adaptor_from_functor{*this, transform}; }
+
+    /*!\brief                 Directly return an instance of the view, initialised with the given parameters.
+     * \param[in] urange      The underlying range.
+     * \param[in] transform   Bool defining whether the range should be reverse complemented.
+     * \returns               A reverse complemented nucleotide range or the original range.
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange, bool const transform) const
+    {
+        static_assert(
+          std::ranges::viewable_range<urng_t>,
+          "The range parameter to views::reverse_complement_or_not cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::sized_range<urng_t>,
+                      "The range parameter to views::reverse_complement_or_not must model std::ranges::sized_range.");
+        static_assert(
+          std::ranges::random_access_range<urng_t>,
+          "The range parameter to views::reverse_complement_or_not must model std::ranges::random_access_range.");
+        static_assert(alphabet::nucleotide_alphabet<std::ranges::range_reference_t<urng_t>>,
+                      "The range parameter to views::reverse_complement_or_not must be over elements of "
+                      "bio::alphabet::nucleotide_alphabet.");
```

Comment:
What I meant was, that the asserts 1 and 3 have a different style from 2 and 4. I am not sure why this happens, because the clang-format file does have `BreakStringLiterals: true`, so I think it should take care of this. 

Anyway, don't worry about it. I will investigate this myself!

Reply:
I tried to fix it but the clang format always threw an error 🤔